up42/storage.py

```diff
@@ -78,6 +78,34 @@ def _query_paginated(
             results_list += response_json["content"]
         return results_list[:limit]
 
+    def _query_paginated_stac_search(
+        self, 
+        url: str, 
+        stac_search_parameters: dict,
+        ) -> list:
+        """
+        Helper to fetch list of items in paginated stac search endpoind, e.g. stac search assets.
+
+        Args:
+            url (str): The base url for paginated endpoint.
+            stac_search_parameters (dict): the parameters required for stac search
+
+        Returns:
+            List of storage STAC results features.
+        """
+        response_features = []
+        while True:
+            stac_results = self.auth._request(
+                request_type="POST", url=url, data=stac_search_parameters
+            )
+            response_features.extend(stac_results["features"])
+            link_elements = [link["body"]["token"] for link in stac_results["links"] if link["rel"] == "next"]
+            if (len(link_elements) > 0):
+                stac_search_parameters["token"] = link_elements[0]
+                continue
+            else:
+                return response_features
```

Comment:
return should be outside of the loop, how about:

        while True:
            stac_results = self.auth._request(
                request_type="POST", url=url, data=stac_search_parameters
            )
            response_features.extend(stac_results["features"])
            token_list = [link["body"]["token"] for link in stac_results["links"] if link["rel"] == "next"]
            if not token_list:
                break
            stac_search_parameters["token"] = link_elements[0]

     return response_features